Engines/Wine/Engine/Versions/script.js

```diff
@@ -1,8 +1,139 @@
+                bVersionNumbers[i] = 0;
+            }
+        }
+
+        // ensure that cases where the description is not set are handled correctly
+        let aVersionDescription = aVersionParts[3];
+        if (typeof aVersionDescription === 'undefined') {
+            aVersionDescription = "";
+        }
+
+        let bVersionDescription = bVersionParts[3];
+        if (typeof bVersionDescription === 'undefined') {
+            bVersionDescription = "";
+        }
+
+        // major
+        if (aVersionNumbers[0] != bVersionNumbers[0]) {
+            return aVersionNumbers[0] - bVersionNumbers[0];
+        }
+        // minor
+        if (aVersionNumbers[1] != bVersionNumbers[1]) {
+            return aVersionNumbers[1] - bVersionNumbers[1];
+        }
+        // patch
+        if (aVersionNumbers[2] != bVersionNumbers[2]) {
+            return aVersionNumbers[2] - bVersionNumbers[2];
+        }
+        // description
+        if (aVersionDescription < bVersionDescription) {
+            return -1;
+        } else {
+            return 1;
+        }
+    }
+    );
+}
+
+/**
+ * Fetches the latest available Wine version based on the category and version regex
+ * @param {wizard} wizard setup wizard to show the download progress of the versions json
+ * @param {string} category (e.g. "upstream-linux-x86")
+ * @param {string} regex regex the version must fulfill
+ * @returns {void}
+ */
+function getLatestVersion(wizard, category, regex) {
+    const versionsJson = module.getAvailableVersions(wizard);
+
+    const packages = versionsJson
+        .filter(distribution => distribution.name === category)
+        .flatMap(distribution => distribution.packages);
+
+    const regExp = new RegExp(regex);
+    const versions = packages
+        .filter(({version}) => regExp.test(version))
+        .flatMap(package => package.version);
+
+    return versions[versions.length-1];
+}
+
+/**
+ * Fetches the available Wine versions (sorted)
+ * @param {wizard} wizard setup wizard to show the download progress
+ * @returns {object} available Wine versions
+ */
+module.getAvailableVersions = function (wizard) {
+    const wineEnginesDirectory = propertyReader.getProperty("application.user.engines") + "/wine";
+    const versionsFile = wineEnginesDirectory + "/availableVersions.json";
+
+    touch(versionsFile);
+
+    const wineWebServiceUrl = propertyReader.getProperty("webservice.wine.url");
+
+    new Downloader()
+        .wizard(wizard)
+        .message(tr("Fetching available Wine versions..."))
+        .url(wineWebServiceUrl)
+        .to(versionsFile)
+        .onlyIfUpdateAvailable(true)
+        .get();
+
+    const versionsJson = JSON.parse(cat(versionsFile));
+
+    versionsJson.forEach(distribution => sortVersions(distribution.packages));
+
+    return versionsJson;
+}
+
+
+module.getLatestStableVersion = function (wizard) {
+    return getLatestVersion(wizard, "upstream-linux-x86", /^\d+\.0(\.\d+)?$/);
```

Comment:
Should we forward the `category` too?

Reply:
When calling `getLatestStableVersion`? I don't think so. What I've thought about is: we could pass the Wine architecture. Normally x86 and amd64 should be updated at the same time. What might maybe happen: the x86 version has just been built but the amd64 not yet. I will add this.